### PR TITLE
test: add shared procedures playground

### PR DESCRIPTION
**Pull request by BeksOmega**
## The basics

- [X] I branched from develop
- [X] My pull request is against develop
- [X] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)
- [X] I ran `npm run format` and `npm run lint`

### Resolves

Blocked by #6475 cuz I need to change the `additionalScripts` to point to the built msgs.

### Proposed Changes

Adds a test page (which doesn't do anything yet) for eventually testing the shared procedures code.

### Reason for Changes

Having a test page is going to be helpful for things that are more difficult to unit test.

### Test Coverage

This is testing silly!

### Documentation

Noup

### Additional Information

N/A

### CODE REVIEW

**Review by cpcallen — APPROVED** [review]
I was expecting to see some code relating to actually sharing procedures here, but as it is it seems to just create two workspaces.  I suppose it can't be written yet because the APIs it depends on don't exist yet.

But it make me wonder: can this not be combined with the existing `multi_playground.html`?  Is there some reason we need multiple multi-`Workspace` playgrounds?  (And with ~~four~~ eight workspaces you can test more complex sharing behaviour, too…)

File: tests/playgrounds/shared_procedures.html [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+// Wait for Blockly to finish loading.
+import '../bootstrap_done.mjs';
+
+const IS_UNCOMPRESSED = !window.bootstrapInfo.compressed;  // See bootstrap.js
+var workspace = null;
+
+function start() {
+  setBackgroundColour();
+
+  var toolbox = document.getElementById('toolbox');
```
[the remark this answers is not in the document]
Comment (BeksOmega) [reply tests/playgrounds/shared_procedures.html:28]: Nope! Just missed this, will fix o7

**Comment by BeksOmega** [comment]
> But it make me wonder: can this not be combined with the existing `multi_playground.html`?  Is there some reason we need multiple multi-`Workspace` playgrounds?  (And with ~~four~~ eight workspaces you can test more complex sharing behaviour, too…)

I think it makes more sense to keep this separate and minimal for now. That way it's clear what code is important for sharing procedures. And I don't feel bad about being messy for testing heheh. But when I'm cleaning this up at the end of the quarter, I'd be happy to merge it with the existing multi playground code.

File: tests/playgrounds/shared_procedures.html [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    <category name="Other Blocks" categorystyle="logic_category">
+      <block type="controls_if"></block>
+      <block type="logic_operation"></block>
+      <block type="logic_negate"></block>
+      <block type="logic_boolean"></block>
+      <block type="text_print"></block>
+    </category>
+    <category name="Variables" categorystyle="variable_category" custom="VARIABLE"></category>
+    <category name="Functions" categorystyle="procedure_category" custom="PROCEDURE"></category>
+  </xml>
```
Comment (cpcallen) [inline tests/playgrounds/shared_procedures.html:190]: Should this be deleted now?

**Comment by BeksOmega** [comment]
Fixes #6512

**Pull request merged**